kubechain/internal/controller/agent/agent_controller.go

```diff
@@ -189,6 +220,27 @@ func (r *AgentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 			statusUpdate.Status.StatusDetail = err.Error()
 			statusUpdate.Status.ValidTools = validTools
 			statusUpdate.Status.ValidMCPServers = validMCPServers
+			statusUpdate.Status.ValidHumanContactChannels = validHumanContactChannels
+			r.recorder.Event(&agent, corev1.EventTypeWarning, "ValidationFailed", err.Error())
+			if updateErr := r.Status().Update(ctx, statusUpdate); updateErr != nil {
+				logger.Error(updateErr, "Failed to update Agent status")
+				return ctrl.Result{}, fmt.Errorf("failed to update agent status: %v", err)
+			}
+			return ctrl.Result{}, err // requeue
+		}
+	}
+
+	// Validate HumanContactChannel references, if any
+	if len(agent.Spec.HumanContactChannels) > 0 {
+		validHumanContactChannels, err = r.validateHumanContactChannels(ctx, &agent)
+		if err != nil {
+			logger.Error(err, "HumanContactChannel validation failed")
+			statusUpdate.Status.Ready = false
+			statusUpdate.Status.Status = StatusError
+			statusUpdate.Status.StatusDetail = err.Error()
+			statusUpdate.Status.ValidTools = validTools
+			statusUpdate.Status.ValidMCPServers = validMCPServers
```

Comment:
weird that this falls through here - like why don't we set it where we compute it? Right now, if I'm reading this code, I have to go scroll up and see where `validMCPServers` came from in order to know what's happening here.

like is this nil in this case? is this computed?